src/Repositories/Interfaces/IIncomeRepository.cs

```diff
@@ -3,6 +3,4 @@
 namespace ExpenseTrackerGroup3.Repositories.Interfaces;
 
 public interface IIncomeRepository : IRepository<Income>
-{
-    Task<Income?> GetMonthlyIncomeByUserId(Guid userId, DateTime month);
```

Comment:
That cannot be so, it is necessary to maintain this method, it is good to use `LINQ` operations, but if we used full operations of this style, the `Repository Pattern` would be redundant, and dependency injection would not make sense.
If data access is performed directly in the service using LINQ queries over a general collection, this principle would be violated and the Repository would be redundant.

Reply:
Having something functional, or something that meets the requirements by breaking the structure to use Linq is not the right thing from my point of view, the abstraction that we are applying with the repositories is done with this purpose of maintaining the work of the queries to the repositories, and also there cannot be an interface without method signatures. In a certain way, a certain coupling is being created between the repositories and the services.

If you really want to keep it like this, I don't have a problem, but before approving it, I would like to know a more appropriate or clear reason than just having something functional.

Reply:
From my point of view I realized that we were handling it.
I think it would be better to do it on a code base so that if we need future modifications we do not depend so much on sql implementations.